projects/graph/src/graph.py

```diff
@@ -5,5 +5,37 @@
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
+
     def __init__(self):
-        pass  # TODO
+        self.vertices = dict()
+
+    def add_vertex(self, vertex):
+        self.vertices[vertex] = set()
+
+    def add_edge(self, num1, num2):
+        if num1 in self.vertices and num2 in self.vertices:
+            self.vertices[num1].add(num2)
+            self.vertices[num2].add(num1)
+            return
+        print("Invalid edges")
+
+    def searchPath(self, sourceVextix, searchItem):
+        # breadth first search
+        visited = []
+        queue = []
+        if sourceVextix == searchItem:
+            return True
+        visited.append(sourceVextix)
+
+        for key in self.vertices[sourceVextix]:
+            queue.append(key)
+        while(len(queue) is not 0):
+            currentKey = queue.pop(0)
+            visited.append(currentKey)
+            if currentKey == searchItem:
+                return True
+            for childkey in self.vertices[currentKey]:
+                if childkey not in visited:
+                    queue.append(childkey)
+        return False
+  
```

Comment:
Nice job with the bfs! I think you are right about not being able to find the shortest path with dfs, but it is useful for finding connected components in a graph. In this case, we don't actually need the optimal result, we just need to be able to find one with the available amount of memory: https://cs.stackexchange.com/questions/73686/why-do-we-prefer-dfs-to-find-connected-components